common/flatpak-utils.c

```diff
@@ -5532,48 +5532,56 @@ flatpak_format_choices (const char **choices,
   g_print ("\n");
 }
 
-/* In this NULL means don't care about these paths, while
-   an empty array means match anything */
 char **
-flatpak_subpaths_merge (char **subpaths1,
-                        char **subpaths2)
+flatpak_strv_merge (char   **strv1,
+                    char   **strv2,
+                    gboolean sort)
 {
   GPtrArray *array;
   int i;
 
   /* Maybe either (or both) is unspecified */
-  if (subpaths1 == NULL)
-    return g_strdupv (subpaths2);
-  if (subpaths2 == NULL)
-    return g_strdupv (subpaths1);
-
-  /* Check for any "everything" match */
-  if (subpaths1[0] == NULL)
-    return g_strdupv (subpaths1);
-  if (subpaths2[0] == NULL)
-    return g_strdupv (subpaths2);
+  if (strv1 == NULL)
+    return g_strdupv (strv2);
+  if (strv2 == NULL)
+    return g_strdupv (strv1);
 
   /* Combine both */
   array = g_ptr_array_new ();
 
-  for (i = 0; subpaths1[i] != NULL; i++)
+  for (i = 0; strv1[i] != NULL; i++)
     {
-      if (!flatpak_g_ptr_array_contains_string (array, subpaths1[i]))
-        g_ptr_array_add (array, g_strdup (subpaths1[i]));
+      if (!flatpak_g_ptr_array_contains_string (array, strv1[i]))
+        g_ptr_array_add (array, g_strdup (strv1[i]));
     }
 
-  for (i = 0; subpaths2[i] != NULL; i++)
+  for (i = 0; strv2[i] != NULL; i++)
     {
-      if (!flatpak_g_ptr_array_contains_string (array, subpaths2[i]))
-        g_ptr_array_add (array, g_strdup (subpaths2[i]));
+      if (!flatpak_g_ptr_array_contains_string (array, strv2[i]))
+        g_ptr_array_add (array, g_strdup (strv2[i]));
     }
 
-  g_ptr_array_sort (array, flatpak_strcmp0_ptr);
-  g_ptr_array_add (array, NULL);
+  if (sort)
```

Comment:
This sort argument is weird. Seems better to sort in flatpak_subpaths_merge. Especially since the behaviour right now means it only sorts in the merge case, not in the early returns, which is right for pre-sorted args, but not necessarily generally.